src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -43,40 +43,68 @@ public QueryTimeouts(IDocumentStore documentStore, string endpointName)
             // default return value for when no results are found
             var nextTimeToRunQuery = DateTime.UtcNow.AddMinutes(10);
 
+            RavenQueryStatistics statistics;
+
             using (var session = documentStore.OpenAsyncSession())
             {
-                var query = GetChunkQuery(session)
-                    .Where(t => t.Time > startSlice)
-                    .Select(t => new
-                    {
-                        t.Id,
-                        t.Time
-                    });
+                int totalCount, skipCount = 0;
 
-                var qhi = new Reference<QueryHeaderInformation>();
-                using (var enumerator = await session.Advanced.StreamAsync(query, qhi).ConfigureAwait(false))
+                do
                 {
-                    while (await enumerator.MoveNextAsync().ConfigureAwait(false))
+                    if (CancellationRequested())
                     {
-                        var dateTime = enumerator.Current.Document.Time;
-                        nextTimeToRunQuery = dateTime; // since results are sorted on time asc, this will get the max time < now
+                        return new TimeoutsChunk(Enumerable.Empty<TimeoutsChunk.Timeout>(), nextTimeToRunQuery);
+                    }
 
-                        if (dateTime > DateTime.UtcNow)
-                        {
-                            break; // break on first future timeout
-                        }
+                    var query = GetChunkQuery(session);
+
+                    var dueTimeouts = await
+                        query.Statistics(out statistics)
+                            .Where(t => t.Time >= startSlice)
```

Comment:
@colin-higgins the v5 code is heavily relying on the `>` instead of `>=`. Because of some changes in v6 it's probably not changing much, but it could probably cause duplicate timeouts. I think I need to go through the new v6 timeout polling to see whether `>=` would still be an issue.

thanks for the background Colin :+1: